<launch>

    <!--Left arm launch files-->
  
    <arg name="arm1" default="l"/>
    <arg name="verbose" default="True" />

    <arg name="mpc_config" default="$(find hrl_haptic_mpc)/mpc_params_pr2.yaml" />
    <arg name="robot_config" default="$(find hrl_multimodal_anomaly_detection)/params/pr2_config_params_l.yaml" />

    <rosparam command="load" file="$(arg mpc_config)" />
    <rosparam command="load" file="$(arg robot_config)" />

    <node pkg="pr2_arm_kinematics" type="pr2_arm_kinematics_node" name="pr2_left_arm_kinematics" output="screen">
      <param name="tip_name" value="l_wrist_roll_link" />
      <param name="root_name" value="torso_lift_link" />
    </node>
    <node name="waypoint_generator" pkg="hrl_haptic_mpc" type="waypoint_generator.py" args="-r pr2 -s fabric -a $(arg arm1)" output="screen"/>
    <node name="robot_haptic_state" pkg="hrl_haptic_mpc" type="robot_haptic_state_node.py" args="-r pr2 -s fabric -a $(arg arm1)" output="screen"/>
    <node name="haptic_mpc" pkg="hrl_haptic_mpc" type="haptic_mpc.py" args="-r pr2 -a $(arg arm1) -v $(arg verbose)" output="screen"/>
  

    <!--Right arm launch files-->

  <!--group ns="right">
    <arg name="arm2" default="r"/>

    <arg name="mpc_config_right" default="$(find hrl_haptic_mpc)/mpc_params_pr2.yaml" />    
    <arg name="robot_config_right" default="$(find hrl_multimodal_anomaly_detection)/params/pr2_config_params_r.yaml" />

    <rosparam command="load" file="$(arg mpc_config_right)" />
    <rosparam command="load" file="$(arg robot_config_right)" />


    <node pkg="pr2_arm_kinematics" type="pr2_arm_kinematics_node" name="pr2_right_arm_kinematics" output="screen">
      <param name="tip_name" value="r_wrist_roll_link" />
      <param name="root_name" value="torso_lift_link" />
    </node>
    <node name="waypoint_generator" pkg="hrl_haptic_mpc" type="waypoint_generator.py" args="-r pr2 -s fabric -a $(arg arm2)" output="screen"/>
    <node name="robot_haptic_state" pkg="hrl_haptic_mpc" type="robot_haptic_state_node.py" args="-r pr2 -s fabric -a $(arg arm2)" output="screen"/>
    <node name="haptic_mpc" pkg="hrl_haptic_mpc" type="haptic_mpc.py" args="-r pr2 -a $(arg arm2) -v $(arg verbose)" output="screen"/>
  </group-->

</launch>
